File: src/d1hashmaps/Java/Main.java
public class Main {
    public static void main(String[] args) {
        DesignHashMap hashMap = new DesignHashMap();
        int[] keys = {5, 2069, 2070, 2073, 4138, 2068};
        int[] values = {100, 200, 400, 500, 1000, 5000};
        String[] operations = {"Get", "Get", "Put", "Get", "Put", "Get", "Get", "Remove", "Get", "Get", "Remove", "Get"};
        int[][] operationKeys = {
            {5}, {2073}, {2073, 250}, {2073},
            {121, 110}, {121}, {2068}, {2069}, {2069},
            {2071}, {2071}, {2071},
        };

        // Initialize the hash map with initial keys and values
        for (int i = 0; i < keys.length; i++) {
            hashMap.put(keys[i], values[i]);
        }

        // Execute the operations
        for (int i = 0; i < operations.length; i++) {
            String op = operations[i];
            switch (op) {
                case "Put":
                    System.out.printf("%d.\tPut(%d, %d)\n", i + 1, operationKeys[i][0], operationKeys[i][1]);
                    hashMap.put(operationKeys[i][0], operationKeys[i][1]);
                    break;
                case "Get":
                    System.out.printf("%d.\tGet(%d)\n", i + 1, operationKeys[i][0]);
                    System.out.printf("\tValue returned: %d\n", hashMap.get(operationKeys[i][0]));
                    break;
                case "Remove":
                    System.out.printf("%d.\tRemove(%d)\n", i + 1, operationKeys[i][0]);
                    hashMap.remove(operationKeys[i][0]);
                    break;
            }

            // Print the current state of the hash map
            System.out.println("\nCurrent Hash Map:");
            hashMap.printHashMap();
            System.out.println("-".repeat(100));
        }
    }
}

File: src/d1hashmaps/TS/main.ts
import { DesignHashMap } from './DesignHashMap';

const main = (): void => {
    const hashMap = new DesignHashMap();
    const keys = [5, 2069, 2070, 2073, 4138, 2068];
    const values = [100, 200, 400, 500, 1000, 5000];
    const operations = ["Get", "Get", "Put", "Get", "Put", "Get", "Get", "Remove", "Get", "Get", "Remove", "Get"];
    const operationKeys = [
        [5], [2073], [2073, 250], [2073],
        [121, 110], [121], [2068], [2069], [2069],
        [2071], [2071], [2071]
    ];

    // Initialize the hash map with initial keys and values
    keys.forEach((key, i) => {
        hashMap.put(key, values[i]);
    });

    // Execute the operations
    operations.forEach((op, i) => {
        switch (op) {
            case "Put":
                console.log(`${i + 1}.\tPut(${operationKeys[i][0]}, ${operationKeys[i][1]})`);
                hashMap.put(operationKeys[i][0], operationKeys[i][1]);
                break;
            case "Get":
                console.log(`${i + 1}.\tGet(${operationKeys[i][0]})`);
                console.log(`\tValue returned: ${hashMap.get(operationKeys[i][0])}`);
                break;
            case "Remove":
                console.log(`${i + 1}.\tRemove(${operationKeys[i][0]})`);
                hashMap.remove(operationKeys[i][0]);
                break;
        }

        // Print the current state of the hash map
        console.log("\nCurrent Hash Map:");
        hashMap.printHashMap();
        console.log('-'.repeat(100));
    });
};

main();
